### ISSUE TITLE
Show hide Tabular dependent inputs

### ISSUE THREAD

**Issue by buttflattery** [buttflattery/yii2-formwizard#60]
When using tabular step we need to have an option where we can show hide the dependent inputs inside the steps when they are loaded for example we need to show the city against the state which is coming from the database and we dont have any way to show the hidden `city` dropdown by looking up into the selected state value

**Comment by buttflattery** [comment]
@girishcp The new feature [dependent inputs](https://github.com/buttflattery/yii2-formwizard/wiki/Dependent-Inputs) closes this issue.

**Comment by girishcp** [comment]
@buttflattery I have added the "depends" option in the widget after updated the latest widget but it doesn't work for me. It's not going either in any conditions when i do alert. WIll this also work with the multiple 

  'depends' => [
                            'attribute' => 'experience_selection',
                            'when' => '1',
                            'condition'=>'==',
                            'then' => new JsExpression("function(id){		
alert(1111);		
                                $('#'+id).closest('.form-group').show();
                            }"),
                            'else' => new JsExpression("function(id){	
alert(222);			
                                $('#'+id).closest('.form-group').hide();
                            }"),
                        ],

**Comment by girishcp** [comment]
@buttflattery 
I have added the "depends" option in the widget after updated the latest widget but it doesn't work for me. It's not going either in any conditions when I do alert. 

'depends' => [
'attribute' => 'experience_selection',
'when' => '1',
'condition'=>'==',
'then' => new JsExpression("function(id){
alert(1111);
$('#'+id).closest('.form-group').show();
}"),
'else' => new JsExpression("function(id){
alert(222);
$('#'+id).closest('.form-group').hide();
}"),
],

**Comment by girishcp** [comment]
@buttflattery 

I am getting the following code when i am using the "Depends" option in my widget, like i have a one dropdown and when i am selecting "YES" then i am showing a textarea type field and when i am selecting "No" then other dropdown type field will show. But i am getting an following error while having this code.

SyntaxError: redeclaration of let thenCallback_0update:239:12
note: Previously declared at line 227, column 12

**MY code is**

'other_brief_experience' => [
	      'containerOptions'=>[
                            'class'=>'form-group'
                        ],
                        'options' => [
                            'type' => 'textarea',
                                         
                            'rows' => 2
                        ],
	    'depends' => [
                            'attribute' => 'smedb_delivery_experience_id',
                            'when' => '1',
                            'condition'=>'==',
                            'then' => new JsExpression("function(id){				
                                $('#'+id).closest('.form-group').show();
                            }"),
                            'else' => new JsExpression("function(id){				
                                $('#'+id).closest('.form-group').hide();
                            }"),
                        ],
                    ],
	'experience_none_selected_id' => [
                        'containerOptions'=>[
                            'class'=>'form-group'
                        ],
                        'depends' => [
                            'attribute' => 'smedb_delivery_experience_id',
                            'when' => '0',
                            'condition'=>'==',
                            'then' => new JsExpression("function(id){
                                $('#'+id).closest('.form-group').show();
                            }"),
                            'else' => new JsExpression("function(id){
                                $('#'+id).closest('.form-group').hide();
                            }"),
                        ],
                        'options' => [
                            'type' => 'dropdown',
                           'itemsList' => [1 => 'Yes', 0 => 'No'],
                            'prompt' => 'Please select a value',
                        ]
	    ],

**Comment by girishcp** [comment]
@buttflattery please update on this

**Comment by buttflattery** [comment]
This feature does not show or hide when you change the drop down but it shows or hides the input this when you are **EDITING** and the values are loaded from the database and you want the dependent inputs to show or hide if the relevant input is selected.

You code above will work when you have the model values loaded from the table for showing or hiding them on runtime should be done using the onchange event by yourself .

About the error you are getting i will push a fix for it. @girishcp

**Comment by girishcp** [comment]
Thanks for your response. it's ok to do not show and hide when we select any value.

But its having an issue with the "depends" when we load the page. @buttflattery

**Comment by buttflattery** [comment]
yeah i know and it is fixed, i just need toverify changes and push

**Comment by buttflattery** [comment]
@girishcp pull the latest code and refresh your browser it should work now

**Comment by girishcp** [comment]
@buttflattery  Thanks for the update.

Now I am getting the following error after updating the latest version and also I didn't find anywhere "isEmptySteps" method in the widget.

**Class 'buttflattery\formwizard\assetbundles\bs3\FormWizardAsset' not found**

**Comment by buttflattery** [comment]
i dont think you have provided the exact error message, the `isEmptySteps` is not inside the `FormWizardAsset` file @girishcp

**Comment by buttflattery** [comment]
have you pulled the latest code? can you verify the function `isEmptySteps` inside the `src/traits/WizardTrait.php` ?

**Comment by girishcp** [comment]
@buttflattery Sorry for the late reply.

Now it's working fine after updating the code but there is a new problem occur When I am adding a second item (by add button) in the STEP_TYPE_TABULAR form then it's taking the same condition (depend) from the first item added. Please check this issue.

**Comment by girishcp** [comment]
@buttflattery 

I hope you are safe and healthy.

Did you get a chance to look into this matter?

Thanks

**Comment by buttflattery** [comment]
@girishcp thanks for the concerns, hard times on the world these days, not getting enough time to work on the extension due to official work from home, so we might have to wait on it. better if you add a separate issue so i remember it

**Comment by girishcp** [comment]
@buttflattery Yes, its really hard time for all of us, just safe and stay at home.
Sure, I will add a separate issue to this.
